### PR TITLE
Sb cross entity session fix

### LINKED ISSUE #31166: [BUG] AcceptNextSessionAsync closes all links if CancellationToken is canceled if EnableCrossEntityTransactions is true
### Library name and version

Azure.Messaging.ServiceBus 7.10.0

### Describe the bug

When creating a ServiceBusSessionReceiver (with a ServiceBusClient which has CrossEntityTransactions enabled) using the `AcceptNextSessionAsync` and trying to create another ServiceBusSessionReceiver with `AcceptNextSessionAsync` which gets a CancellationToken with a timeout it is not possible to use the first ServiceBusSessionReceiver because it is closed because of the cancellation of the cancellation token of the second `AcceptNextSessionAsync` call.

### Scenario:
I try to get multiple session receiver in order to read messages in parallel from multiple sessions and have for each session a single transaction. 
So I try to get ServiceBusSessionReceivers for all available sessions (as far as possible). If there are no more sessions in the service bus the `AcceptNextSessionAsync` method will timeout. Because the timeout it too big I want to cancel after a predefined smaller time and continue.
(Code Snippet below does not show this scenario but a minimal version to show the error.)

### Expected behavior

The link of the first ServiceBusSessionReceiver is not closed and can be used.

### Actual behavior

All connections of all ServiceBusSessionReceiver are closed and `ReceiveMessagesAsync` calls fail with a `ServiceBusException` with the reason `SessionLockLost`

```
Unhandled exception. Azure.Messaging.ServiceBus.ServiceBusException: The session lock was lost. Request a new session receiver. (SessionLockLost)
   at Azure.Messaging.ServiceBus.Amqp.AmqpReceiver.ReceiveMessagesAsyncInternal(Int32 maxMessages, Nullable`1 maxWaitTime, TimeSpan timeout, Cancel
lationToken cancellationToken)
   at Azure.Messaging.ServiceBus.Amqp.AmqpReceiver.<>c.<<ReceiveMessagesAsync>b__40_0>d.MoveNext()
--- End of stack trace from previous location ---
   at Azure.Messaging.ServiceBus.ServiceBusRetryPolicy.RunOperation[T1,TResult](Func`4 operation, T1 t1, TransportConnectionScope scope, Cancellati
onToken cancellationToken, Boolean logRetriesAsVerbose)
   at Azure.Messaging.ServiceBus.ServiceBusRetryPolicy.RunOperation[T1,TResult](Func`4 operation, T1 t1, TransportConnectionScope scope, Cancellati
onToken cancellationToken, Boolean logRetriesAsVerbose)
   at Azure.Messaging.ServiceBus.Amqp.AmqpReceiver.ReceiveMessagesAsync(Int32 maxMessages, Nullable`1 maxWaitTime, CancellationToken cancellationTo
ken)
   at Azure.Messaging.ServiceBus.ServiceBusReceiver.ReceiveMessagesAsync(Int32 maxMessages, Nullable`1 maxWaitTime, Boolean isProcessor, Cancellati
onToken cancellationToken)
   at Azure.Messaging.ServiceBus.ServiceBusReceiver.ReceiveMessagesAsync(Int32 maxMessages, Nullable`1 maxWaitTime, CancellationToken cancellationT
oken)
```

### Reproduction Steps

```csharp
string connectionString = "<ConnectionStringHere>";
ServiceBusClient client = new(connectionString, new ServiceBusClientOptions
{
    EnableCrossEntityTransactions = true
});

// Does have a single session:
ServiceBusSessionReceiver serviceBusSessionReceiver = await client.AcceptNextSessionAsync("myqueue").ConfigureAwait(false);
try
{
    // Does not have any more sessions so the CancellationToken is canceled:
    await client.AcceptNextSessionAsync("myqueue", new ServiceBusSessionReceiverOptions(), new CancellationTokenSource(TimeSpan.FromMilliseconds(100)).Token).ConfigureAwait(false);
}
catch (OperationCanceledException ex)
{
    // Ignore
}

// ServiceBusException is thrown with reason SessionLockLost
IReadOnlyList<ServiceBusReceivedMessage> messages = await serviceBusSessionReceiver.ReceiveMessagesAsync(100).ConfigureAwait(false);
Console.WriteLine(messages.Count);
```

### Environment

OS Name:     Windows
OS Version:  10.0.19042
OS Platform: Windows
RID:               win10-x64
Base Path:   C:\Program Files\dotnet\sdk\6.0.202\

Host (useful for support):
  Version: 6.0.4
  Commit:  be98e88c76

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/31166

### CODE REVIEW

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/tests/Transactions/TransactionLiveTests.cs [hunk 1]
```diff
@@ -971,7 +971,7 @@ public async Task CrossEntityTransactionSessionProcessorRollback()
 
             await processorA.StartProcessingAsync();
             await tcs.Task;
-            await processorA.StopProcessingAsync();
+            await processorA.CloseAsync();
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/tests/Transactions/TransactionLiveTests.cs:974]: This test was only incidentally passing before due to the bug being fixed here - the session lock would get released when an accept session call timed out so on L977 we would be able to lock the session.

**Pull request merged**